piqe_ocp_lib/api/crd/ocp_managed_cluster.py

```diff
@@ -0,0 +1,72 @@
+from piqe_ocp_lib.api.resources.ocp_base import OcpBase
+from kubernetes.client.rest import ApiException
+import logging
+from piqe_ocp_lib import __loggername__
+
+logger = logging.getLogger(__loggername__)
+
+
+class OcpManagedCluster(OcpBase):
+    """
+    OcpManagedCluster Class extends OcpBase and encapsulates all methods
+    related to managing Openshift pods.
+    :param kube_config_file: A kubernetes config file.
+    :return: None
+    """
+    def __init__(self, kube_config_file=None):
+        super(OcpManagedCluster, self).__init__(kube_config_file=kube_config_file)
+        self.api_version = 'cluster.open-cluster-management.io/v1'
+        self.kind = 'ManagedCluster'
+        self.ocp_managed_cluster = self.dyn_client.resources.get(api_version=self.api_version, kind=self.kind)
+
+    def get_managed_clusters(self, namespace="default"):
+        """
+        Get list of managed cluster by hub site
+        :param namespace: (str) namespace
+        :return: ManagedCluster api_response on success OR None on failure
+        """
+        api_response = None
+        try:
+            api_response = self.ocp_managed_cluster.get(namespace=namespace)
+        except ApiException as e:
+            logger.error("Exception while getting managed clusters : %s\n", e)
+
+        return api_response
+
+    def get_managed_cluster_names(self, namespace="default"):
+        """
+        Get managed cluster names
+        :param namespace: namespace
+        :return: list of names on success OR Empty list on failure
+        """
+        managed_clusters = list()
+        managed_clusters_response = self.get_managed_clusters(namespace=namespace)
+
```

Comment:
It would be good to check if the response is None before looping it.

Reply:
Done